src/handlers/interventions-processor-handler.ts

```diff
@@ -63,34 +67,68 @@ async function processSQSRecord(itemFailures: SQSBatchItemFailure[], record: SQS
       return;
     }
 
-    if (isTimestampInFuture(recordBody)) {
+    if (isTimestampInFuture(eventTimestampInMs)) {
+      await sendAuditEvent('AIS_INTERVENTION_IGNORED_INFUTURE', userId, {
+        intervention,
+        reason: 'received event is in the future',
+        appliedAt: undefined,
+      });
       itemFailures.push({
         itemIdentifier: record.messageId,
       });
       return;
+    } else {
```

Comment:
I would get rid of all the `else` or `else if` statements and transform them into simple `if` with a `return;` in each block. I find it easier to follow/read. What do you think?

Reply:
I am in two minds.. I think this block has become quite unwieldy, so perhaps a series of validation method that get called in sequence and throw an error if fail would look better - I will refactor it with multiple if statements for now :) but one worth looking into

Reply:
Will create a ticket for refactoring after merging this in